src/lib.rs

```diff
@@ -80,25 +122,135 @@ impl std::fmt::Debug for MockClock {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("MockClock")
             .field("time", &Self::time())
+            .field("system_time", &Self::system_time())
             .finish()
     }
 }
 
 impl MockClock {
-    /// Set the internal clock to this 'Duration'
+    /// Set the internal Instant clock to this 'Duration'
     pub fn set_time(time: Duration) {
         reference::with_time(|t| *t = time);
     }
 
-    /// Advance the internal clock by this 'Duration'
+    /// Advance the internal Instant clock by this 'Duration'
     pub fn advance(time: Duration) {
         reference::with_time(|t| *t += time);
     }
 
-    /// Get the current duration
+    /// Get the current Instant duration
     pub fn time() -> Duration {
         reference::get_time()
     }
+
+    /// Set the internal SystemTime clock to this 'Duration'
+    pub fn set_system_time(time: Duration) {
+        reference::with_system_time(|t| *t = time);
+    }
+
+    /// Advance the internal SystemTime clock by this 'Duration'
+    pub fn advance_system_time(time: Duration) {
+        reference::with_system_time(|t| *t += time);
+    }
+
+    /// Get the current SystemTime duration
+    pub fn system_time() -> Duration {
+        reference::get_system_time()
+    }
+}
+
+/// An error returned from the duration_since and elapsed methods on SystemTime, used to learn how far in the opposite direction a system time lies.
+#[derive(Clone, Debug)]
+pub struct SystemTimeError(Duration);
+
+impl SystemTimeError {
+    pub fn duration(&self) -> Duration {
+        self.0
+    }
+}
+
+impl std::fmt::Display for SystemTimeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "second time provided was later than self")
+    }
+}
+
+impl std::error::Error for SystemTimeError {
+    #[allow(deprecated)]
+    fn description(&self) -> &str {
+        "other time was not earlier than self"
+    }
+}
+
+/// A simple deterministic SystemTime wrapped around a modifiable Duration
+///
+/// This used a thread-local state as the 'wall clock' that is configurable via
+/// the `MockClock`
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct SystemTime(Duration);
+
+impl SystemTime {
+    pub const UNIX_EPOCH: SystemTime = SystemTime(Duration::from_secs(0));
```

Comment:
Being inside the struct prevents me from accessing it via `use` the way I access std::time::UNIX_EPOCH;

I worked around it with

```
#[cfg(test)]
const UNIX_EPOCH: SystemTime = SystemTime::UNIX_EPOCH;
```

Reply:
Oh, one should be with_system_time. copy-pasted error (basically get-nnn is a read-only accessor, and with-nnn are a way to modify the value atomically)

Reply:
> Being inside the struct prevents me from accessing it via `use` the way I access std::time::UNIX_EPOCH;
> 
> I worked around it with
> 
> ```
> #[cfg(test)]
> const UNIX_EPOCH: SystemTime = SystemTime::UNIX_EPOCH;
> ```

Interesting, I didn't know UNIX_EPOCH was at the module level. std has the struct-level one aliased from the module level one. I added this in 07c1ab2172457672d5ff653625607f486628d649